aws_adfs/_duo_authenticator.py

```diff
@@ -173,10 +173,54 @@ def _authentication_result(
                 response.text
             )
         )
-
-    auth_signature = response.json()['response']['cookie']
+    result_url = response.json()['response']['result_url']
+    duo_result_response = _load_duo_result_url(duo_host, result_url, sid, session, ssl_verification_enabled)
+    auth_signature = duo_result_response.json()['response']['cookie']
     return auth_signature
 
+def _load_duo_result_url(
+        duo_host,
+        result_url,
+        sid,
+        session,
+        ssl_verification_enabled
+):
+    result_for_url = 'https://{}'.format(duo_host) + result_url
+    response = session.post(
+        result_for_url,
+        verify=ssl_verification_enabled,
+        headers=_headers,
+        data={
+            'sid': sid
+        }
+    )
+    logging.debug(u'''Request:
+        * url: {}
+        * headers: {}
+    Response:
+        * status: {}
+        * headers: {}
+        * body: {}
+    '''.format(result_for_url, response.request.headers, response.status_code, response.headers,
+               response.text))
+
+    if response.status_code != 200:
+        raise click.ClickException(
+            u'Issues when following the Duo result URL after '
+            u'authentication. The error response {}'.format(
+                response
+            )
+        )
+    json_response = response.json()
+    print 
```

Comment:
Did you mean to leave this print line in here?

Reply:
Whoops, forgot to save that before committing. It's gone in newest commit.